app.js

```diff
@@ -109,12 +110,24 @@ app.use(function (state, emitter) {
       return createHttpBackend(url)
     } else if (protocol === 'hyper') {
       console.info('creating hyperdrive storage for url', url)
-      return createHyperdriveBackend(url, { swarmOpts: config.swarmOpts, debug: true })
+      return createHyperdriveBackend(url, {
+        swarmOpts: config.swarmOpts,
+        ram: createIdbStorage(url),
+        debug: true
+      })
     } else {
       console.warn('missing protocol handler for url', url)
     }
   }
 
+  function createIdbStorage (url) {
+    try {
+      return idbStorage(url)
+    } catch (e) {
+      console.error('random-access-idb failed', e)
+    }
```

Comment:
If we fail, `undefined` will be returned, so `Hyperdrive()` will default to `random-access-memory`.